### PR TITLE
test/metrics: simplify OOM test

### PR DESCRIPTION
**Pull request by kolyshkin**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:

Currently, this test case keeps waiting for the container that was
already killed in case OOM detection has failed. This is a huge (15
minutes) waste of time.

To fix, change the logic to wait for container exit, and then check the
OOM. This also makes the test code more readable.

Also, add some debug info in case the test has failed.

Also, add a 1 second sleep to the container command, which apparently
improves the chances for the test to succeed.

#### Which issue(s) this PR fixes:

None

Related to: https://github.com/cri-o/cri-o/issues/7035

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/metrics.bats [hunk 1]
```diff
@@ -97,21 +97,13 @@ function teardown() {
...
-		OUTPUT=$(crictl inspect --output yaml "$CTR_ID")
-		if [[ "$OUTPUT" == *"OOMKilled"* ]]; then
-			break
-		fi
-		sleep 10
-	done
-	[[ "$OUTPUT" == *"OOMKilled"* ]]
+	# Wait for container to OOM.
+	EXPECTED_EXIT_STATUS=137 wait_until_exit "$CTR_ID"
+	crictl inspect "$CTR_ID" | jq -e '.status.reason == "OOMKilled"'
```
Comment (haircommander) [inline test/metrics.bats:106]: can we also compress the `ctr.bats` `ctr oom` test into this one?

Comment (kolyshkin) [reply test/metrics.bats:106]: @haircommander not sure what do you mean, I can't find any `ctr oom` test.

Ah! This was already done in commit a1c6ae7e4eb98ef0d511946bcf28e34a345a5876

Comment (kolyshkin) [reply test/metrics.bats:106]: I mean, commit  e914f0e15751fe4903c01f64632ec2fb1f60cb41

**Comment by haircommander** [comment]
the images will automatically update once this merges. the update happens daily at some point. I think @sohankunkerkar knows how to manually triggger it

File: test/metrics.bats [hunk 2]
```diff
@@ -97,21 +97,34 @@ function teardown() {
...
-		CNT=$((CNT + 1))
-		OUTPUT=$(crictl inspect --output yaml "$CTR_ID")
-		if [[ "$OUTPUT" == *"OOMKilled"* ]]; then
-			break
+	# Wait for container to OOM.
+	EXPECTED_EXIT_STATUS=137 wait_until_exit "$CTR_ID"
+	if ! crictl inspect "$CTR_ID" | jq -e '.status.reason == "OOMKilled"'; then
+		# The container has exited but it was not OOM-killed.
+		# Provide some details to debug the issue.
+		echo "--- crlctl inspect :: ---"
```
Comment (haircommander) [inline test/metrics.bats:109]: nit: crictl *

**Comment by sohankunkerkar** [comment]
> The images will automatically update once this merges. the update happens daily at some point. I think @sohankunkerkar knows how to manually trigger it

Hmm... it looks like that PR was merged a couple of weeks ago. Ideally, that change should have been picked up by the periodic job on that day itself.

**Comment by sohankunkerkar** [comment]
Built new images for CI. Let's see if this run passes the integration tests.

File: test/metrics.bats (hunk 2 above)
Comment (kolyshkin) [reply test/metrics.bats]: thanks, fixed

**Comment by sohankunkerkar** [comment]
need to address: https://github.com/cri-o/cri-o/actions/runs/5261831100/jobs/9510391799?pr=7036

**Comment by haircommander** [comment]
@kolyshkin given https://github.com/opencontainers/runc/pull/3932 is in 1.8, instead of this should we update CRI-O to have 1.8 and drop the `inactive-or-failed`?

**Comment by kolyshkin** [comment]
> @kolyshkin given [opencontainers/runc#3932](https://github.com/opencontainers/runc/pull/3932) is in 1.8, instead of this should we update CRI-O to have 1.8 and drop the `inactive-or-failed`?

@haircommander I think that would be a second step. This PR still makes lots of sense because without it the test waits 15 minutes for the condition that is not going to happen).

I will change `sleep 1` to `sleep 5` now.

**Comment by sohankunkerkar** [comment]
https://github.com/cri-o/cri-o/actions/runs/5658658156/job/15330451849?pr=7036
this needs to be fixed.

**Review by sohankunkerkar — APPROVED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]
Thanks!

/retest

**Pull request merged**